crates/shared/src/event_handling.rs

```diff
@@ -387,6 +387,11 @@ where
             {
                 match result {
                     Ok(e) => {
+                        tracing::debug!(
+                            "events fetched for block: {:?}, events: {}",
+                            blocks[i],
+                            e.len(),
```

Comment:
Do we need this log for non-empty events?

Reply:
Probably not. But I guess with this one we can also check if node returns also partial list of events (which should be super weird but hey)

Reply:
To save a bit on logging space I think we should not log empty blocks (by virtue of not having this log we know there were none)